### DISCUSSION TITLE
Lock file not updated with rangeStrategy "widen" (Poetry)

### DISCUSSION THREAD

**Discussion by nphilipp** [renovatebot/renovate#27166] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the problem

I have a couple of Python projects with pretty similar Renovate configuration all of them seem to have stopped updating the `poetry.lock` lock files some time in early January.

I did some experiments and it seemed to hinge on `"rangeStrategy": "widen"`, whenever I removed this line, Renovate would update dependencies in `poetry.lock`.

I created a minimal repository exhibiting the issue: https://github.com/nphilipp-playground/studious-adventure

In it, the `poetry.lock` lock file references `decorator==5.1.0` and trying to update it via the Dependency Dashboard issue changes nothing.

Running `poetry lock` in the repository updates it:
```
nils@makake:~/src/fedora-infra/renovate-poetry-widen-lockfile-issue-test (main)> poetry lock
Updating dependencies
Resolving dependencies... (0.1s)

Writing lock file
nils@makake:~/src/fedora-infra/renovate-poetry-widen-lockfile-issue-test (main)> git diff
diff --git a/poetry.lock b/poetry.lock
index 6efd37f..128c2bd 100644
--- a/poetry.lock
+++ b/poetry.lock
@@ -2,16 +2,16 @@
 
 [[package]]
 name = "decorator"
-version = "5.1.0"
+version = "5.1.1"
 description = "Decorators for Humans"
 optional = false
 python-versions = ">=3.5"
 files = [
-    {file = "decorator-5.1.0-py3-none-any.whl", hash = "sha256:7b12e7c3c6ab203a29e157335e9122cb03de9ab7264b137594103fd4a683b374"},
-    {file = "decorator-5.1.0.tar.gz", hash = "sha256:e59913af105b9860aa2c8d3272d9de5a56a4e608db9a2f167a8480b323d529a7"},
+    {file = "decorator-5.1.1-py3-none-any.whl", hash = "sha256:b8c3f85900b9dc423225913c5aace94729fe1fa9763b38939a95226f02d37186"},
+    {file = "decorator-5.1.1.tar.gz", hash = "sha256:637996211036b6385ef91435e4fae22989472f9d571faba8927ba8253acbc330"},
 ]
 
 [metadata]
 lock-version = "2.0"
 python-versions = "^3.12"
-content-hash = "2bee0741a987a3c0af3b778b1825754f41bf8180a6fecd717ecb5f8de1487dad"
+content-hash = "89cb2371dbdaa78a1141d824f105f060ed228b21dfd6b6220e8ff39a179f9b53"
nils@makake:~/src/fedora-infra/renovate-poetry-widen-lockfile-issue-test (main)>
```

### Relevant debug logs

I don’t know which parts are relevant, so pasted the entire log into [this Gist](https://gist.github.com/nphilipp/53511d7946c2adf231aec609056c8581)

### Have you created a minimal reproduction repository?

I have linked to a minimal reproduction in the description above

**Comment by rarkins** [comment]
What's your reasoning for setting rangeStrategy=widen? That seems like a wrong choice for a dependency like `decorator==5.1.0`.

**Reply by nphilipp** [reply]
The dependency is defined as `decorator = "^5.1.0"` in [`pyproject.toml`](https://github.com/nphilipp-playground/studious-adventure/blob/4a05fe1f94b22c6280f98b9fd5c42c311b0de2a9/pyproject.toml#L9), I’d like to see it widened to `decorator = "^5.1.0 || ^6.0.0"` if a new major version becomes available, in the meantime I’d like the lock file to track minor updates.

**Reply by nphilipp** [reply, marked as answer]
Ahh, just noticed that meanwhile [there’s a PR](https://github.com/nphilipp-playground/studious-adventure/pull/2) for updating the lock file, however I don’t see from the [associated log file](https://developer.mend.io/github/nphilipp-playground/studious-adventure/-/job/018db98c-7a83-7fd6-8f30-4ba03ff8a2d5) ([gist](https://gist.github.com/nphilipp/320ab4c3486f056834192bb7356912f6)) why this happened now and not before.
